Add action and event types TOCs in api.md

timosaikkonen opened the pull request:
The API documentation is quite long and skimming through different types of actions and events is pretty difficult. This PR adds TOCs to Actions and Events sections.

tt-savola commented:
Great, thanks!

A couple of problems:
- [x] The links don't work (the anchors should be like `#create_session` instead of `#-create-session-`).
- [x] All indentation has been changed, which creates a big diff unnecessarily.

timosaikkonen committed:
api: add actions and events tocs

timosaikkonen commented:
> * [ ]  All indentation has been changed, which creates a big diff unnecessarily.

Oh yikes, VSC seems to have been a bit overzealous about formatting. Fixed. I did move one action up on purpose since it seemed to me to belong with other queue-related actions.

edit: moved the corresponding response event as well.
